test/e2e/elasticache/cache_subnet_group/smoke.sh

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env bash
+
+THIS_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+ROOT_DIR="$THIS_DIR/../../../.."
+SCRIPTS_DIR="$ROOT_DIR/scripts"
+
+source "$SCRIPTS_DIR/lib/common.sh"
+source "$SCRIPTS_DIR/lib/k8s.sh"
+source "$SCRIPTS_DIR/lib/testutil.sh"
+
+check_is_installed jq
+
+test_name="$( filenoext "${BASH_SOURCE[0]}" )"
+service_name="elasticache"
+ack_ctrl_pod_id=$( controller_pod_id "$service_name")
+debug_msg "executing test: $service_name/$test_name"
+
+aws_resource_name="ack-test-smoke-${service_name}-subnet-gp"
+k8s_resource_name="cachesubnetgroups/$aws_resource_name"
+
+# pre-req: a subnet id to create cache subnet group
+aws_subnet_id="$(aws ec2 describe-subnets --output json | jq -e '.Subnets[0] | .SubnetId')"
+
+if [ $? -eq 4 ] || [ -z $aws_subnet_id ]; then
+    echo "no subnet id found to run the test. Ensure that subnet id is available to run the test."
```

Comment:
In a future commit, feel free to make the above failure message start with "FAIL: " to match the same style used below on line 39 and 44.

Also, feel free to move this check down to the section that starts on line 36 below since this is a "PRE-CHECK" in the same way as the other pre-checks.